core/src/components/searchbar/searchbar.ionic.scss

```diff
@@ -49,25 +60,101 @@
 // -----------------------------------------
 
 .searchbar-clear-button {
-  @include globals.position(50%, globals.$ionic-space-200, null, null);
+  @include globals.position(
+    searchbarVars.$searchbar-ionic-input-clear-icon-size,
+    searchbarVars.$searchbar-ionic-input-clear-icon-size,
+    null,
+    null
+  );
 
-  position: absolute;
+  width: searchbarVars.$searchbar-ionic-input-clear-icon-size;
+  height: searchbarVars.$searchbar-ionic-input-clear-icon-size;
+
+  background-color: transparent;
 
-  width: globals.$ionic-scale-600;
-  height: globals.$ionic-scale-600;
+  font-size: globals.$ionic-font-size-400;
+
+  contain: strict;
+}
+
+// Searchbar Cancel Icon
+// -----------------------------------------
 
-  transform: translateY(-50%);
+.searchbar-cancel-button {
+  /**
+   * The left edge of the cancel button
+   * should align with the left edge
+   * of the back button if the searchbar
+   * is used in a toolbar.
+   */
+  @include globals.position(0, null, null, 9px);
 
   background-color: transparent;
 
   font-size: globals.$ionic-font-size-400;
+}
 
-  contain: strict;
+// Searchbar Search & Clear Icon & Cancel Icon
+// -----------------------------------------
+
+.searchbar-search-icon,
+.searchbar-clear-button,
+.searchbar-cancel-button {
+  position: absolute;
+}
+
+// Clear Icon & Cancel Icon
+// -----------------------------------------
+
+.searchbar-clear-button:focus-visible,
+.searchbar-cancel-button:focus-visible ion-icon {
+  @include globals.border-radius(globals.$ionic-border-radius-100);
+
+  outline: globals.$ionic-border-size-050 globals.$ionic-border-style-solid globals.$ionic-state-focus-1;
+
+  opacity: 1;
 }
 
 // Searchbar in Toolbar
 // -----------------------------------------
 
 :host-context(ion-toolbar) {
-  min-height: globals.$ionic-scale-1000;
+  min-height: globals.$ionic-scale-1200;
+}
+
+// Searchbar States
+// --------------------------------------------------
+
+/* Hover */
+:host(:hover) {
+  --icon-color: #{globals.$ionic-color-neutral-900};
+}
+
+/* Focus */
+:host(.searchbar-has-focus) .searchbar-input {
+  outline: var(--focus-ring-width) globals.$ionic-border-style-solid var(--focus-ring-color);
+}
+
+:host(.searchbar-has-focus) .searchbar-search-icon {
+  display: block;
+}
+
+:host(.searchbar-has-focus) .searchbar-cancel-button,
+:host(.searchbar-should-show-cancel) .searchbar-cancel-button {
+  display: block;
```

Comment:
Why not combine the `:host(.searchbar-has-focus) .searchbar-search-icon` also here?

Reply:
Good point! https://github.com/ionic-team/ionic-framework/pull/29917/commits/781c5330f9d57fb74fda3d8380a091c2452cc706